src/bosh_azure_cpi/lib/cloud/azure/models/disk_id.rb

```diff
@@ -26,103 +25,77 @@ class DiskId
-      @version = version
-      @id = options[:id]
-      @default_resource_group_name = options[:default_resource_group_name]
-    end
-
     def self.create(caching, use_managed_disks, disk_name: nil, resource_group_name: nil, storage_account_name: nil)
-      id = {
-        'disk_name' => disk_name.nil? ? generate_data_disk_name(use_managed_disks) : disk_name,
-        'caching'   => caching
+      id_hash = {
+        DISK_NAME_KEY => disk_name.nil? ? _generate_data_disk_name(use_managed_disks) : disk_name,
+        CACHING_KEY   => caching
       }
-      id['resource_group_name'] = resource_group_name unless resource_group_name.nil?
-      id['storage_account_name'] = storage_account_name unless storage_account_name.nil?
-      new(VERSION2, id: id)
-    end
-
-    def self.parse(id, azure_config)
-      disk_id = nil
-
-      if id.include?(';')
-        id_hash = {}
-        array = id.split(';')
-        array.each do |item|
-          ret = item.match('^([^:]*):(.*)$')
-          id_hash[ret[1]] = ret[2]
-        end
-        disk_id = new(VERSION2, id: id_hash, default_resource_group_name: azure_config.resource_group_name)
-      else
-        disk_id = new(VERSION1, id: id, default_resource_group_name: azure_config.resource_group_name)
-      end
-
-      disk_id.validate
-      disk_id
+      id_hash[RESOURCE_GROUP_NAME_KEY] = resource_group_name unless resource_group_name.nil?
+      id_hash[STORAGE_ACCOUNT_NAME_KEY] = storage_account_name unless storage_account_name.nil?
+      new(id_hash)
     end
 
-    def self.generate_data_disk_name(use_managed_disks)
-      "#{use_managed_disks ? MANAGED_DATA_DISK_PREFIX : DATA_DISK_PREFIX}-#{SecureRandom.uuid}"
-    end
-
-    def to_s
-      return @id if @version == VERSION1
-      array = []
-      @id.each { |key, value| array << "#{key}:#{value}" }
-      array.sort.join(';')
-    end
-
-    def resource_group_name
-      return @default_resource_group_name if @version == VERSION1
-      @id.fetch('resource_group_name', @default_resource_group_name)
+    def self.parse(id_str, default_resource_group_name)
+      id_hash, plain_id = ResObjectId.parse_with_resource_group(id_str, default_resource_group_name)
+      obj_id = new(id_hash, plain_id)
+      obj_id.validate
+      obj_id
     end
 
     def disk_name
-      return @id if @version == VERSION1
-      @id['disk_name']
+      return @plain_id unless @plain_id.nil?
+      @id_hash[DISK_NAME_KEY]
     end
 
     def caching
-      cloud_error('This function should only be called for data disks') unless disk_name.start_with?(DATA_DISK_PREFIX, MANAGED_DATA_DISK_PREFIX)
+      invalid = !disk_name.start_with?(DATA_DISK_PREFIX, MANAGED_DATA_DISK_PREFIX)
+      cloud_error('This function should only be called for data disks') if invalid
 
-      return parse_data_disk_caching_v1(@id) if @version == VERSION1
-      @id['caching']
+      return _parse_data_disk_caching_v1(@plain_id) unless @plain_id.nil?
```

Comment:
as you removed keyword `v1, v2`, this function should not have `v1` in it anymore. it is a little confusing.

Reply:
good suggestion. use _plain instead.